Artifacts/windows-deploymentgroup/InstallAndConfigureDeploymentGroupAgent.ps1

```diff
@@ -144,6 +144,76 @@ function Configure-DeploymentGroupAgent
 
     if ($runAsAutoLogon)
     {
+        # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+        if ($windowsLogonAccount.Split("\").Count -eq 2)
+        {
+            $domain = $windowsLogonAccount.Split("\")[0]
+        }
+        else
+        {
+            $domain = Hostname
+        }
+        $computerName = "localhost"
+        $password = ConvertTo-SecureString $windowsLogonPassword -AsPlainText -Force
```

Comment:
Where do you validate that the parameter `$windowsLogonPassword` is not `$null` or empty? If you call `ConvertTo-SecureString` with a `$null` value it will throw. I see that we're checking for such condition later on in this flow, but you should consider rearranging that logic.

Reply:
Added a check for this. The later bit of code that checks and adds the password to the command string is common to both flows (autologon or service) hence leaving it as it is.